### PR TITLE
:lipstick: oecd: add metadata for private grants

### PR DESCRIPTION
**Pull request by paarriagadap**
Following comment https://github.com/owid/owid-issues/issues/1620#issuecomment-2431805867

### CODE REVIEW

File: etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml [hunk 1]
```diff
@@ -178,6 +180,38 @@ definitions:
...
+    - "{definitions.key_constant_usd}"
+    - "{definitions.key_oda_list_of_countries}"
+
+  description_key_private_grants: &description_key_private_grants
+    - "{definitions.key_net_private_grants}"
+    - "{definitions.key_private_reporting}"
+    - "{definitions.key_net_disbursements}"
+    - "{definitions.key_constant_usd}"
+
+  description_key_by_donor: &description_key_by_donor
```
Comment (paarriagadap) [inline etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml:209]: Is it possible to avoid this error, @Marigold?

Comment (Marigold) [reply etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml:209]: This was a tough one. The thing is that you cannot use YAML's anchors and references in combination with Jinja templating. This is further complicated by the fact that `description_key` is a list 🤦.

I committed a hacky solution which seems to be working, but is not great. I'll add this example to a [tracking issue](https://github.com/owid/etl/pull/3198) for improving templating experience.

Comment (paarriagadap) [reply etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml:209]: The only issue I see is that both `description_short` and `description_key` are shown in quotes. Is there a way to solve that?
image

Comment (Marigold) [reply etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml:209]: I removed double quotes from contents of `description_short_oda_net_disbursements`. If it is a multi-line string (`|-`) then these quotes are unnecessary.

Comment (paarriagadap) [reply etl/steps/data/garden/oecd/2024-08-21/official_development_assistance.meta.yml:209]: It works! Thank you again! I am not sure why this also fixed description_key, but I am happy anyway.

**Pull request merged**